client/src/cb-admin/pages/Settings.js

```diff
@@ -147,7 +158,91 @@ export default class SettingsPage extends React.Component {
       errors: {},
       dropzoneMsg: undefined,
     });
-  }
+  };
+
+  createZip = () => {
```

Comment:
Think about what you're trying to do here. You have three asynchronous operations, 
1. fetch visitors
2. fetch visits
3. construct csv

One or more of those operations have dependencies on the others:
* (1) depends on nothing
* (2) depends on nothing
* (3) depends on (1) and (2)

This means (3) must be scheduled _after_ (1) and (2) are complete.

The way that you've structured your code at the moment means that (3) is scheduled to happen after (2), but (1) happens in parallel to (2):
```
(1) ------------
(2) ---- (3) ---
```
Starting (1) and (2) at the same time is fine because they don't have any mutual dependencies, but (3) needs to wait until they're both finished before starting. How might we accomplish that?

Reply:
If you're still stuck, look into `Promise.all`

Reply:
Awesome, thank you will do